abci/checktx/mempool_parity_check_tx.go

```diff
@@ -98,10 +106,99 @@ func (m MempoolParityCheckTx) CheckTx() CheckTx {
 			}
 		}
 
+		sdkCtx := m.GetContextForTx(req)
+		lane, err := m.matchLane(sdkCtx, tx)
+		if err != nil {
+			m.logger.Debug("failed to match lane", "lane", lane, "err", err)
+			return sdkerrors.ResponseCheckTxWithEvents(
+				err,
+				0,
+				0,
+				nil,
+				false,
+			), nil
+		}
+
+		consensusParams := sdkCtx.ConsensusParams()
+		laneSize := lane.GetMaxBlockSpace().MulInt64(consensusParams.GetBlock().GetMaxBytes()).TruncateInt64()
+
+		txSize := int64(len(req.Tx))
+		if txSize > laneSize {
+			m.logger.Debug(
+				"tx size exceeds max block bytes",
+				"tx", tx,
+				"tx size", txSize,
+				"max bytes", laneSize,
+			)
+
+			return sdkerrors.ResponseCheckTxWithEvents(
+				fmt.Errorf("tx size exceeds max block bytes"),
+				0,
+				0,
+				nil,
+				false,
+			), nil
+		}
```

Comment:
Does it make sense to do this check before the entire ante-handler sequence is run?

Reply:
I'm not sure - I think the case with a huge Tx is much less likely than the case where youre Tx just fails the antehandler.  So i would rather run that and have the matching logic done later only when we need to